File: processweb/src/main/java/com/pulmwine/processweb/ui/WebActivity.java
package com.pulmwine.processweb.ui;

import android.app.Service;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;
import android.view.KeyEvent;
import com.pulmwine.processweb.R;
import com.pulmwine.processweb.common.WebConstants;
import com.pulmwine.processweb.ui.base.BaseWebFragment;

/**
 * Created by xud on 2018/10/31
 */
public class WebActivity extends AppCompatActivity {

    private String title;
    private String url;

    BaseWebFragment webviewFragment;

    public static void start(Context context, String title, String url, int testLevel) {
        Intent intent = new Intent(context, WebActivity.class);
        intent.putExtra(WebConstants.INTENT_TAG_TITLE, title);
        intent.putExtra(WebConstants.INTENT_TAG_URL, url);
        intent.putExtra("level", testLevel);
        if (context instanceof Service) {
            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
        }
        context.startActivity(intent);
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_common_web);
        title = getIntent().getStringExtra(WebConstants.INTENT_TAG_TITLE);
        url = getIntent().getStringExtra(WebConstants.INTENT_TAG_URL);
        setTitle(title);
        FragmentManager fm = getSupportFragmentManager();
        FragmentTransaction transaction = fm.beginTransaction();

        int level = getIntent().getIntExtra("level", WebConstants.LEVEL_BASE);
        webviewFragment = null;
        if (level == WebConstants.LEVEL_BASE) {
            webviewFragment = CommonWebFragment.Companion.newInstance(url);
        } else {
            webviewFragment = AccountWebFragment.Companion.newInstance(url);
        }
        transaction.replace(R.id.web_view_fragment, webviewFragment).commit();
    }

    @Override
    public boolean onKeyDown(int keyCode, KeyEvent event) {
        if (webviewFragment != null) {
            boolean flag = webviewFragment.onKeyDown(keyCode, event);
            if (flag) {
                return true;
            }
        }
        return super.onKeyDown(keyCode, event);
    }
}

File: CommonWebView/src/main/java/com/plumcookingwine/webview/ui/WebActivity.kt
package com.plumcookingwine.webview.ui

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import com.plumcookingwine.webview.R

class WebActivity : AppCompatActivity() {

    private var fragment: WebFragment? = null

    private var mTitle: String? = null

    private var mUrl: String? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_web)
        mTitle = intent?.getStringExtra("title")
        mUrl = intent?.getStringExtra("url")

        if (fragment == null) {
            fragment = WebFragment.newInstance(mUrl, mTitle)
            supportFragmentManager.beginTransaction()
                .add(R.id.mRlContainer, fragment!!, WebFragment::class.java.simpleName)
                .commitNow()
        }
    }

    override fun onBackPressed() {
        if (fragment?.canGoBack() == false) {
            super.onBackPressed()
        }
    }
}
